artemis-protocols/artemis-amqp-protocol/src/main/java/org/apache/activemq/artemis/protocol/amqp/converter/CoreAmqpConverter.java

```diff
@@ -159,6 +160,12 @@ public static AMQPMessage fromCore(ICoreMessage coreMessage) throws Exception {
          } catch (ActiveMQAMQPIllegalStateException e) {
             properties.setMessageId(messageId);
          }
+      } else {
+         if (message.getInnerMessage().getUserID() != null) {
+            properties.setMessageId("ID:" + message.getInnerMessage().getUserID().toString());
+         } else {
+            properties.setMessageId("ID: " + UUIDGenerator.getInstance().generateStringUUID());
```

Comment:
But JMS does allow a message to **not** have a message ID using the disableMessageId option on the producer.  Not all the clients might support this option but it they did then generating a new ID seems to go against the grain of this issue which is to meet expectations albeit in this case in the negative.

Reply:
Yes that's true. If the core message doesn't have a userID, it should be kept as null.
I'll remove the line.

Reply:
There should be a test for that. Lets say, https://invent.kde.org/snippets/280

The results of that test on this PR at this moment are

```
x [sender=OPENWIRE,consumer=<anything>]
x [sender=CORE,consumer=OPENWIRE]
✓ [sender=CORE,consumer=CORE]
✓ [sender=CORE,consumer=AMQP]
x [sender=AMQP,consumer=OPENWIRE]
✓ [sender=AMQP,consumer=CORE]
✓ [sender=AMQP,consumer=AMQP]
```